demos/service-workflow/src/service.ts

```diff
@@ -0,0 +1,59 @@
+import express, { Express, Request, Response } from 'express';
+import bodyParser from 'body-parser';
+
+//   call this via HTTP GET with a URL like:
+//   http://localhost:9999/get-spanish-greeting?name=Tina
+//   http://localhost:9999/get-spanish-farewell?name=Tina
+
+const app: Express = express();
+const port = 9999;
+
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+
+app.use('/get-spanish-greeting', (req: Request, res: Response) => {
+  if (!req.query.name) {
+    res.status(400);
+    res.send(
+      'Missing required name parameter. Please add ?name=tina to the end of the url.'
+    );
+    return;
```

Comment:
returning after checking input instead of doing an `else` block is much more common